bluetooth: host: split bluetooth.h into gap.h and assigned_numbers.h

PavelVPV opened the pull request:
This PR splits `bluetooth.h` by moving Assigned Numbers define to a new header `assigned_numbers.h` and GAP functionality to `gap.h`.

PavelVPV committed:
bluetooth: host: move assigned numbers out of gap.h

Move Bluetooth Assigned Numbers to own header file to separate them from
core GAP functionality.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

PavelVPV committed:
bluetooth: host: move GAP functionality to gap.h

Move GAP functionality from bluetooth.h to gap.h where it belongs to.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

Thalley commented:
> This PR splits `bluetooth.h` by moving Assigned Numbers define to a new header `assigned_numbers.h` and GAP functionality to `gap.h`.

Do we want to move all the LE Audio related assigned numbers into `assigned_numbers.h` as well?

PavelVPV committed:
bluetooth: host: audio: move audio assigned numbers to dedicated header

Move assigned number for audio from audio.h to assigned_numbers.h where
all assigned numbers are located.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

Thalley commented on include/zephyr/bluetooth/assigned_numbers.h:
```diff
@@ -0,0 +1,1465 @@
+
+/** @} */
+
+/**
+ * @name Generic Audio Assigned Numbers
+ * @{
+ */
```
Do we want to (try to) order the content of this file to match the order in the assigned numbers document? Not sure if the document changes ordering which would make this impossible.

Thalley commented on include/zephyr/bluetooth/assigned_numbers.h:
```diff
@@ -0,0 +1,1465 @@
+ * @}
+ */
+
+/**
+ * @name Appearance Numbers.
+ *
+ * Last Modified on 2023-01-05
+ * @{
```
Should there be a super group for `Core` for this? 

We could in theory have the same structure as the assigned numbers document if we wanted (but also something we can easily add later)

Thalley reviewed (dismissed):
At some point we might want to go through the tree and update the includes if we want to try to keep it IWYU, but I also fully understand why that was not done as part of this, already large, PR :) 

I like the idea of splitting it up like this, which makes it easier to compare with the official documents from the Bluetooth SIG

jhedberg reviewed (changes requested):
Looks generally good. Just one comment regarding copyrights.

PavelVPV commented on include/zephyr/bluetooth/assigned_numbers.h:
```diff
@@ -0,0 +1,1465 @@
+ */
+
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
```
Done. Not sure if I've added them correctly though.

PavelVPV committed:
bluetooth: mesh: move mesh assigned numbers to dedicated header

Move assigned number for mesh from mesh.h to assigned_numbers.h where
all assigned numbers are located.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

alwa-nordic commented:
Splitting out the assigned numbers is a great idea. I like that.
 
For the second commit, I wonder whether this is the way to do it. This PR is wholesale putting basically everything in `bluetooth.h` into `gap.h`, except `bt_enable`, `bt_disable` and `bt_is_ready`. It feels like there is not much point to this.

I think we should instead use `gap.h` and a `bt_gap` namespace to make a distinction between low-level API (e.g. start an advertiser with specific parameters), that can *be used to implement* GAP, and e.g. *actually named* modes and procedures in GAP (Core 3.C.9), like GAP General Discoverable mode.

I realize what I want is a bit fuzzy, but I hope I get the point across?

PavelVPV commented:
> Splitting out the assigned numbers is a great idea. I like that.
> 
> For the second commit, I wonder whether this is the way to do it. This PR is wholesale putting basically everything in `bluetooth.h` into `gap.h`, except `bt_enable`, `bt_disable` and `bt_is_ready`. It feels like there is not much point to this.
> 
> I think we should instead use `gap.h` and a `bt_gap` namespace to make a distinction between low-level API (e.g. start an advertiser with specific parameters), that can _be used to implement_ GAP, and e.g. _actually named_ modes and procedures in GAP (Core 3.C.9), like GAP General Discoverable mode.
> 
> I realize what I want is a bit fuzzy, but I hope I get the point across?

This seems outside the original scope we discussed, estimated, and agreed on. If you don't see much point in this change, I suggest to close this PR and re-evaluate the points you mentioned.
